lib/mix/lib/mix/tasks/release.ex

```diff
@@ -0,0 +1,916 @@
+  For instance, you can use Docker to locally run the same operating
+  system that you have in your production servers, allowing you to
+  assemble a release or build a container or image locally, which is
+  then pushed to production. One can also build images and/or containers
+  at the end of their CI/CD pipelines.
+
+  In other words, there are multiple ways systems can be deployed and
+  releases can be automated and incorporated into all of them as long
+  as you remember that the release must be assembled in the same
+  operating system distribution and version as the production server.
+
+  Once a system is deployed, shutting down said systems can be done by
+  sending SIGINT/SIGTERM to the system, which is what most containers,
+  platforms and tools do, or by explicitly invoking `bin/RELEASE_NAME stop`.
+  Shutting down the system consists of stopping each application and
+  their respective supervisiont trees, one by one, in the opposite order
+  that they were started.
+
+  ## Options
+
+  Inside each release, here are the built-in options:
+
+    * `:applications` - a keyword list that configures and adds new applications
+      to the release. The key is the application name and the value is one of:
+
+        * `:permanent` - the application is started and the node shuts down
+          if the application terminates, regardless of reason
+        * `:transient` - the application is started and the node shuts down
+          if the application terminates abnormally
+        * `:temporary` - the application is started and the node does not
+          shut down if the application terminates
+        * `:load` - the application is only loaded
+        * `:none` - the application is part of the release but it isn't loaded nor
+          started
+
+      All applications default to `:permanent`.
+
+      It defaults to the current application and then we proceed to include all
+      applications the current application depends on recursively. You can
+      include new applications or change the mode of existing ones by listing
+      them here.
+
+      Releases assembled from an umbrella project require this configuration
+      to be explicitly given.
+
+    * `:strip_beams` - a boolean that controls if BEAM files should have their debug
+      information, documentation chunks and other non-essential metadata removed.
+      Defaults to `true`.
+
+    * `:include_executables_for` - a list of atoms with executables to include on
+      the release
+
+    * `:cookie` - a string representing the Erlang Distribution cookie. If no cookie
+      is set, one is automatically generated when the first release is assembled. The
+      cookie will be written to `releases/COOKIE` and shared across multiple release
+      versions. If you are setting this option manually, we recommend the cookie option
+      to be a long and randomly generated, such as:
+      `Base.url_encode64(:crypto.strong_rand_bytes(40))`
```

Comment:
The presence of `-` in the cookie can cause some problems in my experience, all of the following actually:

- `!, ~, -, +, ', ", \, #`

Obviously not all of these are produced by `url_encode64`, but they _are_ characters that are interpreted specially in `vm.args` or on the command line, and had to be filtered out in Distillery at various points. I'd make sure they aren't produced by the tooling here, but should also be documented so that people don't use them in cookies they produce on their own.